src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -222,212 +226,168 @@ private static void handleCommandLineArgs(final String[] args) {
-      ClientLogger.logQuietly(e);
+  private static boolean argsStartsWithPropertykey(String firstArg) {
+    boolean startsWithPropertyKey = false;
+    for (final String arg : COMMAND_LINE_ARGS) {
+      if (firstArg.startsWith(arg)) {
+        startsWithPropertyKey = true;
+        break;
+      }
     }
+    return startsWithPropertyKey;
   }
 
-  private static void checkForMemoryXMX() {
-    final String memSetString = System.getProperty(TRIPLEA_MEMORY_SET, "false");
-    final boolean memSet = Boolean.parseBoolean(memSetString);
-    // if we have already set the memory, then return.
-    // (example: we used process runner to create a new triplea with a specific memory)
-    if (memSet) {
-      return;
-    }
-    final Properties systemIni = getSystemIni();
-    if (useDefaultMaxMemory(systemIni)) {
-      return;
-    }
-    if (getUseMaxMemorySettingOnlyForOnlineJoinOrHost(systemIni)) {
-      return;
-    }
-    long xmx = getMaxMemoryFromSystemIniFileInMB(systemIni);
-    // if xmx less than zero, return (because it means we do not want to change it)
-    if (xmx <= 0) {
-      return;
-    }
-    final int mb = 1024 * 1024;
-    xmx = xmx * mb;
-    final long currentMaxMemory = Runtime.getRuntime().maxMemory();
-    System.out.println("Current max memory: " + currentMaxMemory + ";  and new xmx should be: " + xmx);
-    final long diff = Math.abs(currentMaxMemory - xmx);
-    // Runtime.maxMemory is never accurate, and is usually off by 5% to 15%,
-    // so if our difference is less than 22% we should just ignore the difference
-    if (diff <= xmx * 0.22) {
-      return;
+  private static String getValue(final String arg) {
+    final int index = arg.indexOf('=');
+    if (index == -1) {
+      return "";
     }
-    // the difference is significant enough that we should re-run triplea with a larger number
-    TripleAProcessRunner.startNewTripleA(xmx);
-    // must exit now
-    System.exit(0);
-  }
-
-  public static boolean useDefaultMaxMemory(final Properties systemIni) {
-    final String useDefaultMaxMemoryString = systemIni.getProperty(TRIPLEA_MEMORY_USE_DEFAULT, "true");
-    final boolean useDefaultMaxMemory = Boolean.parseBoolean(useDefaultMaxMemoryString);
-    return useDefaultMaxMemory;
+    return arg.substring(index + 1);
   }
 
-  public static long getMaxMemoryInBytes() {
-    final Properties systemIni = getSystemIni();
-    final String useDefaultMaxMemoryString = systemIni.getProperty(TRIPLEA_MEMORY_USE_DEFAULT, "true");
-    final boolean useDefaultMaxMemory = Boolean.parseBoolean(useDefaultMaxMemoryString);
-    final String maxMemoryString = systemIni.getProperty(TRIPLEA_MEMORY_XMX, "").trim();
-    // for whatever reason, .maxMemory() returns a value about 12% smaller than the real Xmx value.
-    // Just something to be aware of.
-    long max = Runtime.getRuntime().maxMemory();
-    if (!useDefaultMaxMemory && maxMemoryString.length() > 0) {
+  public static void setupLogging(GameMode gameMode) {
```

Comment:
setupLogging was also in the headless version of GameRunner. Same fork strategy was applied here. Diff is still a bit difficult to decipher..